pkg/proxy/register.go

```diff
@@ -116,6 +119,14 @@ func (p *Register) createDirector(proxyDefinition *Definition, balancer Balancer
 			}
 		}
 
+		paramNames := paramNameExtractor.Extract(path)
+		parametrizedPath, err := applyParameters(path, paramNames, req)
```

Comment:
👍

Reply:
here we should have `h.applyParameters` for private funcs

Reply:
I mean let's make it private rather than global

Reply:
Ah and if possible invert the order of params to have the stronger dependencies first e.g `h.applyParameters(req, path, paramNames)`.